packages/wrangler/src/type-generation.ts

```diff
@@ -28,8 +141,11 @@ export async function generateTypes(
 	}
 
 	if (configToDTS.vars) {
-		for (const varName in configToDTS.vars) {
-			const varValue = configToDTS.vars[varName];
+		// Note: vars get overridden by secrets, so should their types
+		const vars = Object.entries(configToDTS.vars).filter(
+			([key]) => !(key in configToDTS.secrets)
+		);
```

Comment:
Could we simplify this by doing something like:

```
const vars = Object.assign({}, configToDTS.vars, configToDTS.secrets);
```

or 

```
const vars = {...configToDTS.vars, ...configToDTS.secrets};
```


Then you don't need to add the secrets below either, right?

Reply:
As I mentioned in my comment above: https://github.com/cloudflare/workers-sdk/pull/5042/files#r1497428862

vars get written as const (`e.g. MY_VAR: "my variable"`) while secrets should definitely be kept generic (`e.g. MY_SECRET: string`)

That's why here I am filtering out the secrets and not combining them with the vars